ethereum_history_api/circuits/lib/src/merkle_patricia_proofs/proof.nr

```diff
@@ -102,7 +104,7 @@ pub fn extract_hash<MAX_PREFIXED_KEY_NIBBLE_LEN>(
     key_ptr: &mut u64
 ) -> Fragment<HASH_LEN> {
     let mut hash = Fragment::new_focused([0; HASH_LEN]);
-    let node_rlp_list: RlpList<MAX_NODE_FIELDS> = decode_list_of_small_strings(node);
+    let node_rlp_list: RlpList<MAX_NODE_FIELDS> = decode_list(node);
```

Comment:
Is there a good reason, why we are using more expensive method here?

Reply:
I thought that in an extension node in the second field we can have an arbitrarily long string, but we can only have a hash there so you're right that we can use `decode_list_of_small_strings`. I'll change it